full-service/src/service/sync.rs

```diff
@@ -222,172 +222,93 @@ pub fn sync_account_next_chunk(
         }
         let end_block_index = end_block_index.unwrap();
 
-        let num_received_txos = if account.view_only {
+        let (view_private_key, account_key) = if account.view_only {
```

Comment:
whether or not we have access to a full `AccountKey` was the main difference between the two previously large blocks of text

Reply:
thanks for taking this on. The places where view account logic was largely duplicative was a bit of tech debt that is wonderful to be addressing.